cypress/tests/data-source.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { DATA_FEDERATION_NAMESPACE } from '../constants/common';
+import { TEST_DATA_SOURCE } from '../constants/tests';
+import {
+  checkDataSourceCreation,
+  createDataSource,
+  deleteDataSourceResources,
+  deleteUsingDetailsPageKebabMenu,
+  navigateToCreatePage,
+  Providers,
+} from '../views/data-resource';
+import { MCGMSCommon } from '../views/mcg-ms-common';
+
+describe('tests for data source creation', () => {
+  before(() => {
+    cy.login();
+    cy.visit('/');
+  });
+
+  beforeEach(() => {
+    MCGMSCommon.visitDataSourceListPage();
+    navigateToCreatePage();
+  });
+
+  afterEach(() => {
+    deleteUsingDetailsPageKebabMenu(TEST_DATA_SOURCE);
+    deleteDataSourceResources(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
+  });
+
+  after(() => {
+    cy.logout();
+  });
+
+  it('test creating data source having AWS as the provider', () => {
```

Comment:
Just a tiny nit: no need to prepend every test with the word "test ..." as the method name (`it`) is the beginning of the description i. e.: `it('creates a data source having ...`.